pkg/safeguards/safeguards.go

```diff
@@ -0,0 +1,311 @@
+package safeguards
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/open-policy-agent/opa/rego"
+	"gopkg.in/yaml.v3"
+)
+
+// Constants
+const ConstraintsDirectory = "./constraints/"
+const DeploymentFilePrefix = "./deployment/"
+
+// ConstraintFetcher is the interface used to fetch each safeguards constraint
+type ConstraintFetcher interface {
+	Fetch() ([]ConstraintFile, map[string]map[string]interface{}, error)
+}
+
+// ConstraintFile is our struct implementation of the safeguards constraint YAML
+// create a getParameters() method
+type ConstraintFile struct {
+	Metadata Metadata `yaml:"metadata"`
+	Spec     Spec     `yaml:"spec"`
+	Name     string
+}
+type Spec struct {
+	Targets []Target `yaml:"targets"`
+}
+type Target struct {
+	Target string   `yaml:"target"`
+	Rego   string   `yaml:"rego"`
+	Libs   []string `yaml:"libs"`
+}
+type Metadata struct {
+	Name string `yaml:"name"`
+}
+
+// ConstraintsBuilderA is the implementation of ConstraintFetcher that reads in constraints from the local fs
+type ConstraintsBuilderA struct {
```

Comment:
rename to `FilesystemConstraintFetcher`?

Reply:
done